add compatibility for django 4.2.1 and Python 3.8.10 using a patch code

JulianDroske opened the pull request:
django and Python have changed over the years. Many modules have been found vunlerable and need to be updated. In order to improve the compatibility and safety without changing the codebase too much, OpenCLMS/compatches is added. For now, it may work (i have no database to test, so, i.e., temporarily).

Commit:
add compatibility for django 4.2.1 and Python 3.8.10

django and Python have changed over the years. Many modules have been found vunlerable and need to be updated.
In order to improve the compatibility and safety without changing the codebase too much, i added /compatches.
For now, it works, temporarily.

xiaohuanshu commented:
LGTM, i will check it later, are you student in gengdan?

JulianDroske commented:
> LGTM, i will check it later, are you student in gengdan?

Yes, a sophomore now. I tried to add as your WeChat friends but had no reply :(
Anyway, I found an possible improvement which displays all courses without conflicts or overlaps. My schedule has an IELTS lesson and a major lesson at the same time, former covers latter. At the moment i need debugging to let it work, so I make this patch first. :)

xiaohuanshu commented:
I didn't receive any friend requests from WeChat. Would you like to try to add it again?

Commit:
apply patches to upgrade to django 4 and python 3

this project is created using some very, very old softwares containing a huge amount of vulnerabilities and
thus cause many problems nowadays without upgrades. these patches make this system support higher version of
base environments and drop support for olds to fix it. testing env. are django 4.2.1 and python 3.8.10.
again, i have no idea on how to fully test it without database, so mark this commit as a "concept".
